File: Domain/Model/ValueObjects/CodeBlock.swift
import Foundation

/// Value object representing a code block with language and content
struct CodeBlock {
    let language: String
    let code: String

    init(language: String, code: String) {
        self.language = language.trimmingCharacters(in: .whitespacesAndNewlines)
        self.code = code
    }

    /// Check if code block is empty
    var isEmpty: Bool {
        return code.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }

    /// Get language display name
    var displayLanguage: String {
        switch language.lowercased() {
        case "python", "py":
            return "Python"
        case "javascript", "js":
            return "JavaScript"
        case "typescript", "ts":
            return "TypeScript"
        case "java":
            return "Java"
        case "cpp", "c++":
            return "C++"
        case "swift":
            return "Swift"
        case "go":
            return "Go"
        case "rust", "rs":
            return "Rust"
        default:
            return language.capitalized
        }
    }
}

// Make it Equatable
extension CodeBlock: Equatable {
    static func == (lhs: CodeBlock, rhs: CodeBlock) -> Bool {
        return lhs.language == rhs.language && lhs.code == rhs.code
    }
}

// Make it Hashable
extension CodeBlock: Hashable {
    func hash(into hasher: inout Hasher) {
        hasher.combine(language)
        hasher.combine(code)
    }
}

// Make it Codable
extension CodeBlock: Codable {}

File: Domain/Model/ValueObjects/CodeBlock.cs
using System;
using System.Collections.Generic;

namespace InterviewMasterApp.Domain.Model
{
    /// <summary>
    /// Value object representing a code block with language and content.
    /// Migrated from Swift: CodeBlock.swift
    /// </summary>
    public class CodeBlock
    {
        public string Language { get; }
        public string Code { get; }

        public CodeBlock(string language, string code)
        {
            Language = (language ?? "text").Trim();
            Code = code ?? "";
        }

        /// <summary>
        /// Check if code block is empty.
        /// </summary>
        public bool IsEmpty => string.IsNullOrWhiteSpace(Code);

        /// <summary>
        /// Get language display name.
        /// </summary>
        public string DisplayLanguage
        {
            get
            {
                return Language.ToLower() switch
                {
                    "python" or "py" => "Python",
                    "javascript" or "js" => "JavaScript",
                    "typescript" or "ts" => "TypeScript",
                    "java" => "Java",
                    "cpp" or "c++" => "C++",
                    "csharp" or "c#" => "C#",
                    "swift" => "Swift",
                    "go" => "Go",
                    "rust" or "rs" => "Rust",
                    "kotlin" => "Kotlin",
                    _ => char.ToUpper(Language[0]) + Language.Substring(1)
                };
            }
        }

        public override bool Equals(object? obj)
        {
            return obj is CodeBlock other &&
                   Language == other.Language &&
                   Code == other.Code;
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(Language, Code);
        }

        public override string ToString()
        {
            return $"CodeBlock({DisplayLanguage}, {Code.Length} chars)";
        }
    }
}
